### PR TITLE
Add `npm ls` in test command

### PR DESCRIPTION
**Pull request by allevo**
As titled.

This PR prevents switching among branches with different dependencies.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by jsumners** [comment]
How does this accomplish the stated goal?

**Comment by allevo** [comment]
`npm ls` exits using code `1` if the installed dependencies are not the same that are described into `package.json`

**Comment by jsumners** [comment]
Okay, so how does that help when it is run after the unit tests?

**Comment by allevo** [comment]
@jsumners I you prefer, we can make that check as the first one

**Comment by jsumners** [comment]
Wouldn't that be the point? To make sure the dependencies match what is declared prior to testing the framework?

**Comment by mcollina** [comment]
I am -1 to this change. This increases the time that it will take to run `npm test` in any normal case/workflow.

It is true that it happens from time to time, but tests would fail if the versions are not compatible, right?

I'm ok in having it in the pre-commit hook, because you shouldn't commit something that passes the test with unmet dependencies (but please redirect the output).

**Comment by allevo** [comment]
> but please redirect the output

Is there a way for output redirect that works for unix and windows (at least)?

**Comment by mcollina** [comment]
I'm not sure.

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "typings": "fastify.d.ts",
   "scripts": {
     "lint": "standard | snazzy",
+    "checkdeps": "node -e 'c = child_process.spawnSync(\"npm\", [\"ls\"]); console.log(c.status === 0 ? \"\": c.stderr.toString()); process.exit(c.status)'",
```
Comment (delvedor) [inline package.json]: Just to be sure, will this work also under Windows?

Comment (allevo) [reply package.json]: :smile: 
I have no idea. But I'm using nodejs without doing nothing in bash. I think so.

Anyway if only I need this "feature" we are able to not land it

**Comment by mcollina** [comment]
I'm closing this, fell free to reopen if you still want to pursue.

**Pull request closed without merging**